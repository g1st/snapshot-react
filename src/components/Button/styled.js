import styled from 'styled-components';
import { darken } from 'polished';

import { colorMainButton, colorText } from '../../styles/variables';

export const MainButton = styled.button`
  width: ${props => (props.fullSpan ? '100%' : 'auto')};
  display: inline-block;
  background-color: ${colorMainButton};
  padding: 14px 32px;
  letter-spacing: 3px;
  font-size: 1.4rem;
  display: inline-block;
  font-weight: 300;
  margin: 0 auto;
  color: #fff;
  outline: none;
  text-transform: capitalize;
  border-radius: 5px;
  border: none;
  transition: all 0.3s;
  cursor: pointer;

  :hover {
    color: ${colorText};
    background-color: ${darken(0.5, `${colorMainButton}`)};
  }
`;
MainButton.displayName = 'MainButton';
