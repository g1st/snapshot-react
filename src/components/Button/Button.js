import React from 'react';
import PropTypes from 'prop-types';

import { MainButton } from './styled';

const propTypes = {
  title: PropTypes.string,
  big: PropTypes.bool
};

const defaultProps = { title: 'Click Me', big: true };

const Button = ({ title, big }) => {
  return <MainButton fullSpan={big}>{title}</MainButton>;
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
