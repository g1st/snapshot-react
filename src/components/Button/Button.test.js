import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '.';

describe('Button Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<Button title="Title" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("renders 'Title' if title is passed", () => {
    const tree = shallow(<Button title="Title">Title</Button>);
    expect(tree.childAt(0).text()).toEqual('Title');
  });
  it("renders 'Click Me' if title is missing", () => {
    const tree = shallow(<Button />);
    expect(tree.childAt(0).text()).toEqual('Click Me');
  });
});
