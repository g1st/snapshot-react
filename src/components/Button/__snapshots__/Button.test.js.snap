// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<MainButton
  fullSpan={true}
>
  Title
</MainButton>
`;
